leader/src/jerigon.rs

```diff
@@ -27,21 +27,24 @@ pub(crate) async fn jerigon_main(
     )
     .await?;
 
-    let block_proofs = prover_input
+    // We keep only last block proof from the interval. It contains
+    // all the necessary information to verify the whole interval.
+    let last_block_proof = prover_input
         .prove(&runtime, previous_proof, save_inputs_on_error)
```

Comment:
**design question**: we may want to store / flush intermediate proofs on the go, in case the program aborts for some reasons, so that we don't restart everything from scratch? I don't think the extra I/O would be prohibitive.

Reply:
We need to change logic then and store them the moment they are generated, at the moment all the proofs are collected first and then stored